### ISSUE TITLE
Make popup menu WCGA 2.1 compliant

### ISSUE THREAD

**Issue by nikku** [bpmn-io/diagram-js#871]
### Is your feature request related to a problem? Please describe.

Currently the popup menu is a keyboard trap that is insufficiently indicated to assistive technology (screen readers), it violates [`no-keyboard-trap`](https://www.w3.org/WAI/WCAG22/quickref/#no-keyboard-trap).

### Describe the solution you'd like

The popup editor is clearly indicated as a keyboard trap with `ESC` (cancel) or `ENTER` (to select action) being the two promoted navigation actions.

### Describe alternatives you've considered

None.

### Additional context

Found via https://github.com/camunda/web-modeler/issues/7929#issuecomment-2003338026.

**Comment by nikku** [comment]
I added the following as an alternative, which more clearly follows existing usage patterns (and `no-keyboard-trap`):

> Do not trap TAB but instead only trap arrow keys. Tab will focus individual entries in the select, and tabbing out of the component will close it.

**Comment by barmac** [comment]
Drop `TAB` makes sense to me.

**Comment by nikku** [comment]
#874 ships initial support, we'll iterate if there is more to do.
